presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -226,4 +247,73 @@ void nextAttempt()
             }
         }
     }
+
+    public static final class RetryStats
+    {
+        private final CounterStat success = new CounterStat();
+        private final CounterStat finalFailure = new CounterStat();
+        private final TimeDistribution failureLatency = new TimeDistribution(MICROSECONDS);
+        private final DistributionStat successRetry = new DistributionStat();
+        private final TimeDistribution successLatency = new TimeDistribution(MICROSECONDS);
+
+        @Managed
+        @Nested
+        public CounterStat getSuccess()
+        {
+            return success;
+        }
+
+        @Managed
+        @Nested
+        public CounterStat getFinalFailure()
+        {
+            return finalFailure;
+        }
+
+        @Managed
+        @Nested
+        public TimeDistribution getFailureLatency()
+        {
+            return failureLatency;
+        }
+
+        @Managed
+        @Nested
+        public DistributionStat getSuccessRetry()
```

Comment:
I think it has a meaning of "attempts before success". Let's call it something similar.